site/tests/BaseUnitTest.php

```diff
@@ -95,7 +107,11 @@ function ($permission) use ($access) {
 
         $mock_queries = $this->createMock(DatabaseQueries::class);
         foreach ($queries as $method => $value) {
-            $mock_queries->method($method)->willReturn($value);
+            $this->mocked_methods[$method] = false;
+            $mock_queries->method($method)->will($this->returnCallback(function () use ($method, $value) {
+                $this->mocked_methods[$method] = true;
```

Comment:
Also yeah this is gonna give big merge conflicts so possibly branch off of my branch so that you have these changes and they don't make giant merge conflicts? If you do that though my branch will need to be merged first.

Reply:
Included changes in the latest commit of my [branch](https://github.com/PlasmaIntec/Submitty/tree/HomePageUnitTests)